1. A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years. Ask user for their salary and year of
service and print the net bonus amount

In [1]:

working_period = int(input('enter your working years ='))
salary = int(input('enter your Salary ='))
if working_period >= 5:
    bonus = (salary/100)*5
    print('Congrats your bonus is',bonus )
else :
    print('this offer is not for you')



enter your working years = 5
enter your Salary =50000
Congrats your bonus is 2500.0


2. Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [1]:
age = int(input("Enter Your Age = "))

if age>17 :
    print("You are eligble for Voting")
    
else:
    print("You are not eligible for Voting")

Enter Your Age = 22
You are eligble for Voting


3. Write a program to check whether a number entered by user is even or odd

In [2]:
number = int(input("Enter number you want to check "))

if number%2 == 0 :
    print("The number you enter  is Even Number")
    
else:
    print("The number you enter is Odd Numb")

Enter number you want to check 11
The number you enter is Odd Numb


4. Write a program to check whether a number is divisible by 7 or not. Show Answer.

In [3]:
number = int(input("Enter Number you want to check "))

if number%7 == 0:
    print("Number is Divisible by 7")
    
else:
    print("Number is not Divisible by 7")

Enter Number you want to check 630
Number is Divisible by 7


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [4]:
number = int(input("Enter Number "))

if number%5==0 :
    print("Hello")
else:
    print("Bye")

Enter Number 150
Hello


7) Write a program to display the last digit of a number

In [5]:
number = int(input("Enter a number: "))

last_digit = number % 10
print("The last digit of", number, "is:", last_digit)

Enter a number: 18892
The last digit of 18892 is: 2


8. Write a program to check whether the last digit of a number( entered by user ) is divisible by 3 or not.

In [6]:
number = int(input("Enter a number: "))
last_digit = number % 10
if last_digit % 3 == 0:
    print("last digit of a number is Divisible by 3")
else:
    print("last digit of a number is not Divisible by 3")

Enter a number: 573
last digit of a number is Divisible by 3


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle

In [12]:
length = int(input("Enter length of the shape = "))
breadth = int(input("Enter breadth of the shape = "))


if length == breadth :
    area = length * breadth
    print("It is a Square")
    print("And the area of the Square is =", area, "Sq foot")

else:
    area = length * breadth
    print("It is a Rectangle")
    print("The area of Rectangle is = ",area , "Sq foot")

Enter length of the shape = 20
Enter breadth of the shape = 20
It is a Square
And the area of the Square is = 400 Sq foot


10) Take two int values from user and print greatest among them.

In [16]:
number1 = int(input("Enter First Number = "))
number2 = int(input("Enter Second Number = "))

greater_number = max(number1, number2)

print("The greater number is =",greater_number)


Enter First Number = 45
Enter Second Number = 90
The greater number is = 90


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000. Ask user for quantity Suppose, one unit will cost 100.
Judge and print total cost for user.

In [19]:
purchased_quantity = int(input("Enter the total purchased items = "))
discount = 0.1
unit_cost = 100

if purchased_quantity >1000 :
    print("Congratulation! Your are Eligible for the Discount Offer")
    actual_cost = purchased_quantity * unit_cost
    print("Your actual cost is = ",actual_cost)
    total_cost = purchased_quantity * unit_cost * discount
    print("Your Total cost after discount is =",total_cost)
    
else:
    print("Sorry! Your are Not Eligible for the Discount Offer")
    total_cost =  purchased_quantity * unit_cost 
    print("Your Total cost is = ",total_cost)

Enter the total purchased items = 3500
Congratulation! Your are Eligible for the Discount Offer
Your actual cost is =  350000
Your Total cost after discount is = 35000.0


12) A school has following rules for grading system:
a. Below 25 - F
b. 25 to 45 - E
c. 45 to 50 - D
d. 50 to 60 - C
e. 60 to 80 - B
f. Above 80 - A
Ask user to enter marks and print the corresponding grade.

In [20]:
marks = int(input("Enter Your Marks ="))
if marks >= 80:
    print("Congratulation! You have Secured A grade")
elif marks >= 60:
    print("You have Secured B Grade")
elif marks >= 50:
    print("You have Secured C Grade")
elif marks >= 45:
    print("You have Secured D Grade")
elif marks >= 25:
    print("You have Secured E Grade")
else:
    print("You have Secured F Grade, you are Fail")

Enter Your Marks =90
Congratulation! You have Secured A grade


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.
Take following input from user
Number of classes held
Number of classes attended.
And print
percentage of class attended
Is student is allowed to sit in exam or not

In [22]:
a = int(input("Enter Number of classes Held = "))
b = int(input("Enter Number of classes Attended = "))
attendence_percentage = b/a * 100

if attendence_percentage >75:
    print("Congtatulation! You are allowed to sit in exam")

else:
    print("Sorry! You are not allowed to sit in exam because your attendence percentage is less than 75%")

Enter Number of classes Held = 25
Enter Number of classes Attended = 20
Congtatulation! You are allowed to sit in exam


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and
print accordingly

In [24]:
a = int(input("Enter Number of classes Held = "))
b = int(input("Enter Number of classes Attended = "))
attendence_percentage = b/a * 100
c = input("Any Medical Casue or Not = ")

if attendence_percentage > 75:
    print("Congtatulation! You are allowed to sit in exam")
elif c == "Y":
    print("You are allowed to sit in exam")
    print("But you need to show the proof of your medical issue")
else:
    print("Sorry! You are not allowed to sit in exam because your attendence percentage is less than 75%")

Enter Number of classes Held = 25
Enter Number of classes Attended = 15
Any Medical Casue or Not = Y
You are allowed to sit in exam
But you need to show the proof of your medical issue


16) Write a program to check if a year is leap year or not.
If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [25]:
year = int(input("Enter year = "))

if year%4 == 0:
    print("It is leap Year")
    if year%400 == 0:
        print("It is a Century Year")
    else:
        print("but it is not a century Year")
else:
    print("It is not a Leap year")

Enter year = 2024
It is leap Year
but it is not a century Year


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.
if employee is female, then she will work only in urban areas.
if employee is a male and age is in between 20 to 40 then he may work in anywhere
if employee is male and age is in between 40 t0 60 then he will work in urban areas only.
And any other input of age should print "ERROR"

In [27]:
age = int(input("Enter your age = "))
gender = input("Enter your Gender = ")
marital_status = input("Enter your Marital Status = ")


if gender == "F":
    print("She will work only in urban areas")
elif gender == "M":
        if 20 <= age <= 40:
            print("He may work in anywhere")
        elif 41 <= age <= 60:
            print("He will work in urban areas only") 
        else:
            print("ERROR: invalid age for males")
else:
    print("ERROR: inavlid gender")

Enter your age = 22
Enter your Gender = M
Enter your Marital Status = N
He may work in anywhere


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria : Unit Price
uptp 100 units no charge Next 200 units Rs 5 per unit After 200 units Rs 10 per unit (For example if input unit is 350 than total bill amount is
Rs.3500 (For example if input unit is 97 than total bill amount is Rs.0 (For example if input unit is 150 than total bill amount is Rs.750

In [6]:
units = int(input("Enter the Total number of units = "))
if units <= 100:
    charges = units * 0
    print("Your Electricity Charges for the Period is =",charges)
elif 100 <= units <= 200 :
    charges = units * 5
    print("Your Electricity Charges for the Period is =",charges)
elif units > 200 :
    charges = units * 10
    print("Your Electricity Charges for the Period is =",charges)
else:
    print("Invalid Error")

Enter the Total number of units = 400
Your Electricity Charges for the Period is = 4000


13) Take input of age of 3 people by user and determine oldest and youngest among them

In [7]:
age1 = int(input("Enter age of person 1 : "))
age2 = int(input("Enter age of person 2 : "))
age3 = int(input("Enter age of person 3 : "))

oldest = max(age1, age2, age3)
youngest = min(age1, age2, age3)
print("Oldest person's age :", oldest)
print("Youngest person's age :", youngest)

Enter age of person 1 : 12
Enter age of person 2 : 22
Enter age of person 3 : 40
Oldest person's age : 40
Youngest person's age : 12
